contracts/LimePlace.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "./TokenList.sol";
+
+contract LimePlace {
+    using TokenList for TokenList.List;
+
+    uint256 public LISTING_FEE = 100000000000000 wei;
+    address payable private _marketOwner;
+    mapping(bytes32 => Listing) private _listings;
+    
+    //keep track of active listings
+    TokenList.List private _activeListings;
```

Comment:
Currently, you're using `_activeListings` only to fetch the currently active ones on-chain. This defeats the purpose of the API you're going to build and adds overhead to each function. The API's goal is to exactly serve you information about the listings.